%\VignetteIndexEntry{isomiRs}
%\VignetteKeywords{smallRNA, miRNAs,isomiRs, DifferentailExpression}
%\VignetteEngine{knitr::knitr}
\documentclass{article}
\usepackage[utf8]{inputenc}

<<knitr, echo=FALSE, results="hide">>=
library("knitr")
opts_chunk$set(tidy=FALSE,
            fig.width=5,fig.height=5,
            message=FALSE)
@


<<style, eval=TRUE, echo=FALSE, results="asis">>=
BiocStyle::latex()
@

\title{isomiRs}
\author{Lorena Pantano$^{1*}$, Georgia Escaramis$^{2*}$,
Eulalia Martin$^{2*}$ \\ \small{Harvard H Chan School of Public Health,
Boston, US;} \\ \small{$^{2}$ Center of Genomic Regulation, Barcelona,
Spain;} \\ \small{\texttt{$^*$lpantano (at) iscb.org}}}
\date{Modified: 3 Feb, 2015. Compiled: \today}

\begin{document}

\maketitle
\tableofcontents
\newpage

\section*{Introduction}

miRNA are small RNA fragments (18-23 nt long) that influence gene expression
during development and cell stability. Morin et al \cite{morin}, discovered
isomiRs first time after sequencing human stem cells.

IsomiRs are miRNAs that vary slightly in sequence, which result from variations
in the cleavage site during miRNA biogenesis (5’-trimming and 3’-trimming
variants), nucleotide additions to the 3’-end of the mature miRNA
(3’-addition variants) and nucleotide modifications (substitution variants)\cite{emarti}.

There are many tools designed for isomiR detections, although the majority are
web application where user can not control the analysis. The two main command
tools for isomiRs mapping are SeqBuster and sRNAbench. \Biocpkg{isomiRs}
package is designed to analyze the output of SeqBuster tool or any other
tool after converting to the desire format.

\section{Citing isomiRs}

If you use the package, please cite the associated paper \cite{isomirs}.

\section{Input format}

The input should be a file for each sample in the following format:

<<format-line,eval=FALSE,echo=TRUE>>==
sequence name counts mirna start end mismatches additions 5trimming
3trimming NA NA DB multimapping
@

This is the standard output of SeqBuster tool, but can be converted from
any other tool having the mapping information on the precursors. Read more on
\href{https://github.com/lpantano/seqbuster/wiki/miraligner}{miraligner manual}

\section{IsomirDataSeq class}

This object will store all raw data from the input files and all processed object used for visualization and statistical analysis. It is a is a subclass of \Rfunction{SummarizedExperiment} with the \Rfunction{colData} and \Rfunction{counts} methods. Beside that, the user can access the normalized count matrix with \Rfunction{normcounts}, and the raw data with \Rfunction{isoraw} method. This list contains the input matrix for each sample. 

\subsection{Access data}

The main slots are count data and design matrix.

\section{Quick start}

We are going to use a small RNAseq data coming from human
frontal cortex samples to give some basic examples of isomiRs analyses.

In this data set we will find two groups:
* b: individuals with less than a year
* o: individuals in the elderly.


<<package-load,message=FALSE>>=
library(isomiRs)
data(isomiRexp)
@


\subsection{Reading input}

The function \Rfunction{loadIso} needs a vector with the paths for each file
and a data.frame with the design experiment similar to the ones used for
differential expression.

\subsection{Descriptive analysis}

You can plot isomiRs tendencies with \Rfunction{plotIso}. In this figure you
will have how abundant is each type of isomiRs at different positions
considering the total abundance and the total number of sequences.
With the \Rfunction{type}, controls what type of isomiRs to show:
trimming, addition or substitution.

<<package-plot-iso,message=FALSE>>=
obj<-plotIso(isomiRexp,type="t5")
@

\subsection{Differential expression analysis}

The \Rfunction{deIso} has implemented \Biocpkg{DESeq2} package.
You can using directly from this function. This function has parameters
to create the input for DESeq2 only with reference miRNA, or all isomiRs, or
collapsing all isomiRs together.

<<package-de,message=FALSE>>=
dds<-deIso(obj,formula=~condition)
library(DESeq2)
plotMA(dds)
head(results(dds))
@

You can differentiate between reference sequences, and isomirs at 5' end
doing this:

<<package-de-iso5,message=FALSE>>=
dds = deIso(obj,formula=~condition,ref=TRUE,iso5=TRUE)
head(results(dds))
@

\subsection{Count data}

the \Rfunction{countsIso} can get the matrix counts that can be use for many
different downstream analyses changing the way isomiRs are collapsed.
Similar to the previous point.

<<package-count,message=FALSE>>=
obj = countsIso(obj)
head(counts(obj))
@

The normalization using the \Rfunction{rlog} from \Biocpkg{DESeq2} package
allows quick integration to another analyses like clustering.


<<package-norm,message=FALSE>>=
obj = normIso(obj)
heatmap(counts(obj, norm=TRUE)[1:100,], labRow = "")
@

\subsection{Supervised classification}

Partial Least Squares Discriminant Analysis (PLS-DA) is a technique specifically
appropriate for analysis of high dimensionality data sets and multicollineality
\cite{perezenciso}. PLS-DA is a supervised method (i.e. makes use of class
labels) with the aim to provide a dimension reduction strategy in a situation
where we want to relate a binary response variable (in our case young or old
status) to a set of predictor variables. Dimensionality reduction procedure is
based on orthogonal transformations of the original variables (isomiRs) into a
set of linearly uncorrelated latent variables (usually termed as components)
such that maximizes the separation between the different classes in the first
few components \cite{xia}. We used sum of squares captured by the model (R2) as
a goodness of fit measure. We implemented this method using the
\Rpackage{DiscriMiner}. Moreover, the function \Rfunction{isoPLSDAplot}
summarizes the result of this analysis.


<<package-pls,eval=TRUE>>=
obj = countsIso(obj,iso5=TRUE,iso3=TRUE,add=TRUE,ref=TRUE)
obj = normIso(obj)
pls.obj = isoPLSDA(obj, "condition", nperm = 10)
isoPLSDAplot(pls.obj$component, colData(obj)[,"condition"])
@


The analysis can be done again, using only the most important discriminant
isomiRS from the PLS-DA models based on the analysis. We used Variable
Importance for the Projection (VIP) criterion to select the most important
features, since takes into account the contribution of a specific predictor
for both the explained variability on the response and the explained
variability on the predictors.


<<package-plsplot,message=FALSE,eval=FALSE>>=
pls.obj = isoPLSDA(obj,"condition", refinment = FALSE, vip = 0.8)
@


\pagebreak

%---------------------------------------------------------
\section*{Session info}
%---------------------------------------------------------
Here is the output of \Rfunction{sessionInfo} on the system on which
this document was compiled:

<<sessionInfo,results='asis',echo=FALSE>>=
toLatex(sessionInfo())
@

\bibliography{isomirs}

\end{document}
