<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Characterization of miRNA and isomiR molecules • isomiRs</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">isomiRs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/isomiRs.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Characterization of miRNA and isomiR molecules</h1>
                        <h4 class="author">Lorena Pantano</h4>
            
            <h4 class="date">22 March 2018</h4>
          </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>isomiRs package version: 1.7.2</p>
    </div>
    
<div class="contents">
<p><a href="lorena.pantano@gmail.com">Lorena Pantano</a> Harvard TH Chan School of Public Health, Boston, US</p>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>miRNAs are small RNA fragments (18-23 nt long) that influence gene expression during development and cell stability. Morin et al <span class="citation">(Morin et al. 2008)</span>, discovered isomiRs first time after sequencing human stem cells.</p>
<p>IsomiRs are miRNAs that vary slightly in sequence, which result from variations in the cleavage site during miRNA biogenesis (5’-trimming and 3’-trimming variants), nucleotide additions to the 3’-end of the mature miRNA (3’-addition variants) and nucleotide modifications (substitution variants)<span class="citation">(Martí et al. 2010)</span>.</p>
<p>There are many tools designed for isomiR detection, however the majority are web application where user can not control the analysis. The two main command tools for isomiRs mapping are SeqBuster and sRNAbench <span class="citation">(Guillermo et al. 2014)</span>. <em><a href="http://bioconductor.org/packages/isomiRs">isomiRs</a></em>. package is designed to analyze the output of SeqBuster tool or any other tool after converting to the desire format.</p>
</div>
<div id="citing-isomirs" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-isomirs" class="anchor"></a>Citing isomiRs</h2>
<p>If you use the package, please cite this paper <span class="citation">(Pantano L 2010)</span>.</p>
</div>
<div id="input-format" class="section level2">
<h2 class="hasAnchor">
<a href="#input-format" class="anchor"></a>Input format</h2>
<p>The input should be the output of SeqBuster-miraligner tool (*.mirna files). It is compatible with <a href="http://github.com/mirtop/mirtop">mirTOP</a> tool as well, which parses BAM files with alignments against miRNA precursors.</p>
<p>For each sample the file should have the following format:</p>
<pre><code>seq                     name           freq mir           start end  mism  add  t5  t3
TGTAAACATCCTACACTCAGCT  seq_100014_x23  23  hsa-miR-30b-5p  17  40   0       0  0   GT
TGTAAACATCCCTGACTGGAA   seq_100019_x4   4   hsa-miR-30d-5p  6   26   13TC    0  0   g
TGTAAACATCCCTGACTGGAA   seq_100019_x4   4   hsa-miR-30e-5p  17  37   12CT    0  0   g
CAAATTCGTATCTAGGGGATT   seq_100049_x1   1   hsa-miR-10a-3p  63  81   0       TT 0   ata
TGACCTAGGAATTGACAGCCAGT seq_100060_x1   1   hsa-miR-192-5p  25  47   8GT     0  c   agt</code></pre>
<p>This is the standard output of SeqBuster-miraligner tool, but can be converted from any other tool having the mapping information on the precursors. Read more on <a href="http://seqcluster.readthedocs.org/mirna_annotation.html">miraligner manual</a>.</p>
</div>
<div id="isomirdataseq-class" class="section level2">
<h2 class="hasAnchor">
<a href="#isomirdataseq-class" class="anchor"></a>IsomirDataSeq class</h2>
<p>This object will store all raw data from the input files and some processed information used for visualization and statistical analysis. It is a subclass of <code>SummarizedExperiment</code> with <code>colData</code> and <code>counts</code> methods. Beside that, the object contains raw and normalized counts from miraligner allowing to update the summarization of miRNA expression.</p>
<div id="access-data" class="section level3">
<h3 class="hasAnchor">
<a href="#access-data" class="anchor"></a>Access data</h3>
<p>The user can access the normalized count matrix with <code><a href="../reference/counts.html">counts(object, norm=TRUE)</a></code>.</p>
<p>You can browse for the same miRNA or isomiRs in all samples with <code>isoSelect</code> method.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(isomiRs)
<span class="kw">data</span>(mirData)
<span class="kw">head</span>(<span class="kw"><a href="../reference/isoSelect.html">isoSelect</a></span>(mirData, <span class="dt">mirna=</span><span class="st">"hsa-let-7a-5p"</span>, <span class="dv">1000</span>))</code></pre></div>
<pre><code>## DataFrame with 6 rows and 7 columns
##                                                id       cc1       cc2
##                                       &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;
## 1  hsa-let-7a-5p 0 0 0 0 : TGAGGTAGTAGGTTGTATAGTT    235825    171354
## 2   hsa-let-7a-5p 0 0 0 gtt : TGAGGTAGTAGGTTGTATA      1173       884
## 3   hsa-let-7a-5p 0 0 0 t : TGAGGTAGTAGGTTGTATAGT     50920     52403
## 4 hsa-let-7a-5p 0 0 0 T : TGAGGTAGTAGGTTGTATAGTTT      8806      5478
## 5   hsa-let-7a-5p 0 0 0 tt : TGAGGTAGTAGGTTGTATAG      6041      5467
## 6 hsa-let-7a-5p 0 A 0 0 : TGAGGTAGTAGGTTGTATAGTTA      9616      4742
##         cc3       ct2       ct3       ct4
##   &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## 1    149541    163569    114028    123454
## 2       751       756       495       719
## 3     35525     35795     24993     34512
## 4      5427      6045      4626      4181
## 5      3817      4828      2858      5087
## 6      5437      7326      4790      4430</code></pre>
<p><code>metadata(mirData)</code> contains two lists: <code>rawList</code> is a list with same length than number of samples and stores the input files for each sample; <code>isoList</code> is a list with same length than number of samples and stores information for each isomiR type summarizing the different changes for the different isomiRs (trimming at 3’, trimming a 5’, addition and substitution). For instance, you can get the data stored in <code>isoList</code> for sample 1 and 5’ changes with this code <code>metadata(ids)[["isoList"]][[1]]["t5sum"]</code>.</p>
</div>
<div id="isomirs-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#isomirs-annotation" class="anchor"></a>isomiRs annotation</h3>
<p>IsomiR names follows this structure:</p>
<ul>
<li>miRNA name</li>
<li>type: ref if the sequence is the same than the miRNA reference. <code>iso</code> if the sequence has variations. <em>t5 tag: indicates variations at 5’ position. The naming contains two words: <code>direction - nucleotides</code>, where direction can be UPPER CASE NT (changes upstream of the 5’ reference position) or LOWER CASE NT (changes downstream of the 5’ reference position). <code>0</code> indicates no variation, meaning the 5’ position is the same than the reference. After <code>direction</code>, it follows the nucleotide/s that are added (for upstream changes) or deleted (for downstream changes). </em>t3 tag: indicates variations at 3’ position. The naming contains two words: <code>direction - nucleotides</code>, where direction can be LOWER CASE NT (upstream of the 3’ reference position) or UPPER CASE NT (downstream of the 3’ reference position). <code>0</code> indicates no variation, meaning the 3’ position is the same than the reference. After <code>direction</code>, it follows the nucleotide/s that are added (for downstream changes) or deleted (for upstream chanes). <em>ad tag: indicates nucleotides additions at 3’ position. The naming contains two words: <code>direction - nucleotides</code>, where direction is UPPER CASE NT (upstream of the 5’ reference position). <code>0</code> indicates no variation, meaning the 3’ position has no additions. After <code>direction</code>, it follows the nucleotide/s that are added. </em>mm tag: indicates nucleotides substitutions along the sequences. The naming contains three words: <code>position-nucleotideATsequence-nucleotideATreference</code>. *seed tag: same than <code>mm</code> tag, but only if the change happens between nucleotide 2 and 8.</li>
</ul>
<p>In general nucleotides in UPPER case mean insertions respect to the reference sequence, and nucleotides in LOWER case mean deletions respect to the reference sequence.</p>
</div>
</div>
<div id="reading-input" class="section level2">
<h2 class="hasAnchor">
<a href="#reading-input" class="anchor"></a>Reading input</h2>
<p>We are going to use a small RNAseq data from human brain samples <span class="citation">(<span class="citeproc-not-found" data-reference-id="pantano2016"><strong>???</strong></span>)</span> to give some basic examples of isomiRs analyses.</p>
<p>In this data set we will find two groups:</p>
<p><em>pc: 7 control individuals </em>pt: 7 patients with Parkinson’s Disease in early stage.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(isomiRs)
<span class="kw">data</span>(mirData)</code></pre></div>
<p>The function <code>IsomirDataSeqFromFiles</code> needs a vector with the paths for each file and a data frame with the design experiment similar to the one used for a mRNA differential expression analysis. Row names of the data frame should be the names for each sample in the same order than the list of files.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ids &lt;-<span class="st"> </span><span class="kw"><a href="../reference/IsomirDataSeqFromFiles.html">IsomirDataSeqFromFiles</a></span>(fn_list, <span class="dt">design=</span>de)</code></pre></div>
</div>
<div id="descriptive-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#descriptive-analysis" class="anchor"></a>Descriptive analysis</h2>
<p>You can plot isomiRs expression with <code>isoPlot</code>. In this figure you will see how abundant is each type of isomiRs at different positions considering the total abundance and the total number of sequences. The <code>type</code> parameter controls what type of isomiRs to show. It can be trimming (iso5 and iso3), addition (add) or substitution (subs) changes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ids &lt;-<span class="st"> </span><span class="kw"><a href="../reference/isoCounts.html">isoCounts</a></span>(mirData)
<span class="kw"><a href="../reference/isoPlot.html">isoPlot</a></span>(ids, <span class="dt">type=</span><span class="st">"all"</span>, <span class="dt">column =</span> <span class="st">"group"</span>)</code></pre></div>
<p><img src="isomiRs_files/figure-html/package-plot-iso-t5-1.png" width="672"></p>
</div>
<div id="count-data" class="section level2">
<h2 class="hasAnchor">
<a href="#count-data" class="anchor"></a>Count data</h2>
<p><code>isoCounts</code> gets the count matrix that can be used for many different downstream analyses changing the way isomiRs are collapsed. The following command will merge all isomiRs into one feature: the reference miRNA.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="../reference/counts.html">counts</a></span>(ids))</code></pre></div>
<pre><code>##                    cc1    cc2    cc3    ct2    ct3    ct4
## hsa-let-7a-2-3p      5     13      4      4      2     12
## hsa-let-7a-3p      767    707    630    496    343    635
## hsa-let-7a-5p   317730 244832 203888 222066 154246 175523
## hsa-let-7b-3p     1037   1949    679    931    494   1149
## hsa-let-7b-5p    69671  64702  42347  45071  28312  41214
## hsa-let-7c-3p       36     16     25     30     28     26</code></pre>
<p>The normalization uses <code>rlog</code> from <em><a href="http://bioconductor.org/packages/DESeq2">DESeq2</a></em> package and allows quick integration to another analyses like heatmap, clustering or PCA.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(pheatmap)
ids =<span class="st"> </span><span class="kw"><a href="../reference/isoNorm.html">isoNorm</a></span>(ids, <span class="dt">formula =</span> <span class="op">~</span><span class="st"> </span>group)
<span class="kw"><a href="http://www.rdocumentation.org/packages/pheatmap/topics/pheatmap">pheatmap</a></span>(<span class="kw"><a href="../reference/counts.html">counts</a></span>(ids, <span class="dt">norm=</span><span class="ot">TRUE</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>,], 
         <span class="dt">annotation_col =</span> <span class="kw">data.frame</span>(<span class="kw">colData</span>(ids)[,<span class="dv">1</span>,<span class="dt">drop=</span><span class="ot">FALSE</span>]),
         <span class="dt">show_rownames =</span> <span class="ot">FALSE</span>, <span class="dt">scale=</span><span class="st">"row"</span>)</code></pre></div>
<p><img src="isomiRs_files/figure-html/package-norm-1.png" width="672"></p>
<div id="differential-expression-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#differential-expression-analysis" class="anchor"></a>Differential expression analysis</h3>
<p>The <code>isoDE</code> uses functions from <em><a href="http://bioconductor.org/packages/DESeq2">DESeq2</a></em> package. This function has parameters to create a matrix using only the reference miRNAs, all isomiRs, or some of them. This matrix and the design matrix are the inputs for DESeq2. The output will be a DESeqDataSet object, allowing to generate any plot or table explained in DESeq2 package vignette.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds &lt;-<span class="st"> </span><span class="kw"><a href="../reference/isoDE.html">isoDE</a></span>(ids, <span class="dt">formula=</span><span class="op">~</span>group)
<span class="kw">library</span>(DESeq2)
<span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/plotMA">plotMA</a></span>(dds)</code></pre></div>
<p><img src="isomiRs_files/figure-html/package-de-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/results">results</a></span>(dds, <span class="dt">format=</span><span class="st">"DataFrame"</span>))</code></pre></div>
<pre><code>## log2 fold change (MLE): group ct vs cc 
## Wald test p-value: group ct vs cc 
## DataFrame with 6 rows and 6 columns
##                     baseMean log2FoldChange     lfcSE       stat    pvalue
##                    &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
## hsa-let-7a-2-3p 7.280132e+00     -0.4810538 1.1938136 -0.4029555 0.6869810
## hsa-let-7a-3p   6.037792e+02     -0.4826551 0.4335156 -1.1133513 0.2655576
## hsa-let-7a-5p   2.204578e+05     -0.3924390 0.3990463 -0.9834423 0.3253898
## hsa-let-7b-3p   1.121845e+03     -0.6713768 0.5885204 -1.1407876 0.2539583
## hsa-let-7b-5p   4.932097e+04     -0.6020825 0.4364083 -1.3796311 0.1677003
## hsa-let-7c-3p   2.645655e+01      0.3445930 0.6119527  0.5631040 0.5733640
##                      padj
##                 &lt;numeric&gt;
## hsa-let-7a-2-3p  0.955863
## hsa-let-7a-3p    0.955863
## hsa-let-7a-5p    0.955863
## hsa-let-7b-3p    0.955863
## hsa-let-7b-5p    0.955863
## hsa-let-7c-3p    0.955863</code></pre>
<p>You can differentiate between reference sequences and isomiRs at 5’ end with this command:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds =<span class="st"> </span><span class="kw"><a href="../reference/isoDE.html">isoDE</a></span>(ids, <span class="dt">formula=</span><span class="op">~</span>group, <span class="dt">ref=</span><span class="ot">TRUE</span>, <span class="dt">iso5=</span><span class="ot">TRUE</span>)
<span class="kw">head</span>(<span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/results">results</a></span>(dds, <span class="dt">tidy=</span><span class="ot">TRUE</span>))</code></pre></div>
<pre><code>##                        row    baseMean log2FoldChange     lfcSE       stat
## 1 hsa-let-7a-2-3p.iso.t5:0   3.5706434     -0.5907229 1.5874311 -0.3721251
## 2 hsa-let-7a-2-3p.ref.t5:0   3.6094368     -0.3092342 1.5174726 -0.2037824
## 3   hsa-let-7a-3p.iso.t5:0 419.6204486     -0.3906909 0.4241191 -0.9211821
## 4   hsa-let-7a-3p.iso.t5:A   0.3029715      1.8405090 4.0509840  0.4543363
## 5   hsa-let-7a-3p.iso.t5:c  95.2721196     -0.3703600 0.5018339 -0.7380131
## 6  hsa-let-7a-3p.iso.t5:ct   4.6224205     -1.2606914 1.2513812 -1.0074399
##      pvalue      padj
## 1 0.7097997 0.9832495
## 2 0.8385236 0.9928199
## 3 0.3569554 0.9832495
## 4 0.6495868 0.9832495
## 5 0.4605065 0.9832495
## 6 0.3137234 0.9832495</code></pre>
<p>Alternative, for more complicated cases or if you want to control more the differential expression analysis paramters you can use directly <em><a href="http://bioconductor.org/packages/DESeq2">DESeq2</a></em> package feeding it with the output of <code><a href="../reference/counts.html">counts(ids)</a></code> and <code>colData(ids)</code> like this:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds =<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/DESeq2/topics/DESeqDataSet">DESeqDataSetFromMatrix</a></span>(<span class="kw"><a href="../reference/counts.html">counts</a></span>(ids),
                             <span class="kw">colData</span>(ids), <span class="dt">design =</span> <span class="op">~</span><span class="st"> </span>group)</code></pre></div>
</div>
</div>
<div id="supervised-classification" class="section level2">
<h2 class="hasAnchor">
<a href="#supervised-classification" class="anchor"></a>Supervised classification</h2>
<p>Partial Least Squares Discriminant Analysis (PLS-DA) is a technique specifically appropriate for analysis of high dimensionality data sets and multicollineality <span class="citation">(Pérez-Enciso and Tenenhaus 2003)</span>. PLS-DA is a supervised method (i.e. makes use of class labels) with the aim to provide a dimension reduction strategy in a situation where we want to relate a binary response variable (in our case young or old status) to a set of predictor variables. Dimensionality reduction procedure is based on orthogonal transformations of the original variables (isomiRs) into a set of linearly uncorrelated latent variables (usually termed as components) such that maximizes the separation between the different classes in the first few components <span class="citation">(Xia and Wishart 2011)</span>. We used sum of squares captured by the model (R2) as a goodness of fit measure. We implemented this method using the <code>DiscriMiner</code> into <code>isoPLSDA</code> function. The output p-value of this function will tell about the statistical significant of the group separation using miRNA expression data. Moreover, the function <code>isoPLSDAplot</code> helps to visualize the results. It will plot the samples using the significant components (t1, t2, t3 …) from the PLS-DA analysis and the samples distribution along the components.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ids =<span class="st"> </span><span class="kw"><a href="../reference/isoCounts.html">isoCounts</a></span>(ids, <span class="dt">iso5=</span><span class="ot">TRUE</span>, <span class="dt">minc=</span><span class="dv">10</span>, <span class="dt">mins=</span><span class="dv">6</span>)
ids =<span class="st"> </span><span class="kw"><a href="../reference/isoNorm.html">isoNorm</a></span>(ids, <span class="dt">formula =</span> <span class="op">~</span><span class="st"> </span>group)
pls.ids =<span class="st"> </span><span class="kw"><a href="../reference/isoPLSDA.html">isoPLSDA</a></span>(ids, <span class="st">"group"</span>, <span class="dt">nperm =</span> <span class="dv">2</span>)
df =<span class="st"> </span><span class="kw"><a href="../reference/isoPLSDAplot.html">isoPLSDAplot</a></span>(pls.ids)</code></pre></div>
<p><img src="isomiRs_files/figure-html/package-pls-1.png" width="672"></p>
<p>The analysis can be done again using only the most important discriminant isomiRS from the PLS-DA models based on the analysis. We used Variable Importance for the Projection (VIP) criterion to select the most important features, since takes into account the contribution of a specific predictor for both the explained variability on the response and the explained variability on the predictors.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pls.ids =<span class="st"> </span><span class="kw"><a href="../reference/isoPLSDA.html">isoPLSDA</a></span>(ids,<span class="st">"group"</span>, <span class="dt">refinment =</span> <span class="ot">FALSE</span>, <span class="dt">vip =</span> <span class="fl">0.8</span>)</code></pre></div>
</div>
<div id="gene---mirna-integration" class="section level2">
<h2 class="hasAnchor">
<a href="#gene---mirna-integration" class="anchor"></a>Gene - miRNA integration</h2>
<p>The package offers a correlation analysis of miRNA and gene expression data. Having two SummarizedExperiments objects with their expression, the target prediction for each miRNA, the function <code>isoNetwork</code> and <code>isoPlotNetwork</code> can generate a summarized figure showing the relationship between expression profile, miRNA repression and enrichment analysis:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># library(org.Mm.eg.db)</span>
<span class="co"># library(clusterProfiler)</span>
<span class="kw">data</span>(isoExample)

<span class="co"># ego &lt;- enrichGO(row.names(assay(gene_ex_rse, "norm")),</span>
<span class="co">#                 org.Mm.eg.db, "ENSEMBL", ont = "BP")</span>
data =<span class="st"> </span><span class="kw"><a href="../reference/isoNetwork.html">isoNetwork</a></span>(mirna_ex_rse, gene_ex_rse, ma_ex,
                  <span class="dt">org =</span> ego)</code></pre></div>
<pre><code>## Dimmension of cor matrix: 20 25</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/isoPlotNet.html">isoPlotNet</a></span>(data)</code></pre></div>
<p><img src="isomiRs_files/figure-html/isonetwork-1.png" width="672"></p>
<p>As an option, <code>org</code> can be <code>org.Mm.eg.db</code> and <code>genename</code> can be <code>ENSEMBL</code> and it will run <code>enrcihGO</code> internally.</p>
<p>To create the ma_ex matrix, the user can use <code>findTargets</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mirna_ma &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">rbinom</span>(<span class="dv">20</span> <span class="op">*</span><span class="st"> </span><span class="dv">25</span>, <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="dv">1</span>), <span class="dt">ncol =</span> <span class="dv">20</span>)
<span class="kw">colnames</span>(mirna_ma) &lt;-<span class="st"> </span><span class="kw">rownames</span>(mirna_ex_rse)
<span class="kw">rownames</span>(mirna_ma) &lt;-<span class="st"> </span><span class="kw">rownames</span>(gene_ex_rse)
ma_ex &lt;-<span class="st"> </span><span class="kw"><a href="../reference/findTargets.html">findTargets</a></span>(mirna_ex_rse, gene_ex_rse, mirna_ma)</code></pre></div>
<pre><code>## Dimmension of cor matrix: 20 25</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(ma_ex[,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>])</code></pre></div>
<pre><code>##                    mmu-miR-181c-5p mmu-miR-181a-5p mmu-miR-221-3p
## ENSMUSG00000021822       0.0000000      -0.7333333      0.0000000
## ENSMUSG00000062960       0.0000000       0.0000000     -0.7333333
## ENSMUSG00000032462       0.0000000       0.0000000      0.0000000
## ENSMUSG00000004530      -0.7333333       0.0000000      0.0000000
## ENSMUSG00000026193       0.0000000       0.0000000      0.0000000
## ENSMUSG00000059049       0.0000000       0.0000000      0.0000000
##                    mmu-miR-204-5p
## ENSMUSG00000021822            0.0
## ENSMUSG00000062960            0.0
## ENSMUSG00000032462           -0.6
## ENSMUSG00000004530            0.0
## ENSMUSG00000026193            0.0
## ENSMUSG00000059049            0.0</code></pre>
<p>And to get the <code>mirna_ma</code> data.frame with the miRNA-target information, the user can use <code>mirna2targetscan</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mirna_ma &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mirna2targetscan.html">mirna2targetscan</a></span>(<span class="kw">c</span>(<span class="st">"hsa-miR-34c-5p"</span>))
<span class="kw">head</span>(mirna_ma)</code></pre></div>
<pre><code>##                         miRFamily Seedmatch  PCT  gene
## 1 miR-34ac/34bc-5p/449abc/449c-5p      8mer 0.29 79719
## 2 miR-34ac/34bc-5p/449abc/449c-5p      8mer 0.76   215
## 3 miR-34ac/34bc-5p/449abc/449c-5p   7mer-1a 0.58  3983
## 4 miR-34ac/34bc-5p/449abc/449c-5p   7mer-1a 0.58  3983
## 5 miR-34ac/34bc-5p/449abc/449c-5p   7mer-1a 0.58  3983
## 6 miR-34ac/34bc-5p/449abc/449c-5p   7mer-1a 0.58  3983</code></pre>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<p>Here is the output of <code>sessionInfo</code> on the system on which this document was compiled:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>## R version 3.4.3 (2017-11-30)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] DESeq2_1.19.29             pheatmap_1.0.8            
##  [3] bindrcpp_0.2.0.9000        isomiRs_1.7.2             
##  [5] SummarizedExperiment_1.8.1 DelayedArray_0.4.1        
##  [7] matrixStats_0.52.2         Biobase_2.38.0            
##  [9] GenomicRanges_1.30.1       GenomeInfoDb_1.14.0       
## [11] IRanges_2.12.0             S4Vectors_0.16.0          
## [13] BiocGenerics_0.24.0        DiscriMiner_0.1-29        
## [15] BiocStyle_2.5.41          
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-6              bit64_0.9-7              
##  [3] RColorBrewer_1.1-2        rprojroot_1.2            
##  [5] tools_3.4.3               backports_1.1.2          
##  [7] R6_2.2.2                  rpart_4.1-11             
##  [9] KernSmooth_2.23-15        Hmisc_4.1-1              
## [11] DBI_0.7                   lazyeval_0.2.1           
## [13] colorspace_1.3-2          nnet_7.3-12              
## [15] tidyselect_0.2.3          gridExtra_2.3            
## [17] GGally_1.3.2              bit_1.1-12               
## [19] compiler_3.4.3            htmlTable_1.11.2         
## [21] labeling_0.3              caTools_1.17.1           
## [23] scales_0.5.0              checkmate_1.8.5          
## [25] readr_1.1.1               genefilter_1.60.0        
## [27] stringr_1.2.0             digest_0.6.13            
## [29] foreign_0.8-69            rmarkdown_1.6.0.9008     
## [31] XVector_0.18.0            base64enc_0.1-3          
## [33] pkgconfig_2.0.1           htmltools_0.3.6          
## [35] limma_3.34.7              htmlwidgets_1.0          
## [37] rlang_0.1.6.9003          rstudioapi_0.7           
## [39] RSQLite_2.0               bindr_0.1.0.9000         
## [41] BiocParallel_1.12.0       gtools_3.5.0             
## [43] acepack_1.4.1             dplyr_0.7.4.9000         
## [45] RCurl_1.95-4.8            magrittr_1.5             
## [47] GenomeInfoDbData_0.99.1   Formula_1.2-2            
## [49] Matrix_1.2-12             Rcpp_0.12.15             
## [51] munsell_0.4.3             stringi_1.1.6            
## [53] yaml_2.1.16               zlibbioc_1.24.0          
## [55] gplots_3.0.1              plyr_1.8.4               
## [57] grid_3.4.3                blob_1.1.0               
## [59] gdata_2.18.0              lattice_0.20-35          
## [61] splines_3.4.3             annotate_1.56.0          
## [63] hms_0.3                   locfit_1.5-9.1           
## [65] knitr_1.19                pillar_1.1.0             
## [67] codetools_0.2-15          geneplotter_1.56.0       
## [69] XML_3.98-1.9              glue_1.2.0               
## [71] evaluate_0.10.1           latticeExtra_0.6-28      
## [73] data.table_1.10.4-3       tidyr_0.8.0              
## [75] gtable_0.2.0              purrr_0.2.4              
## [77] reshape_0.8.7             assertthat_0.2.0         
## [79] ggplot2_2.2.1             xtable_1.8-2             
## [81] survival_2.41-3           tibble_1.4.2             
## [83] targetscan.Hs.eg.db_0.6.1 AnnotationDbi_1.40.0     
## [85] memoise_1.1.0             cluster_2.0.6</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-barturen2014">
<p>Guillermo, Barturen, Rueda Antonio, Hamberg Maarten, Alganza Angel, Lebron Ricardo, Kotsyfakis Michalis, Shi BuJun, KoppersLalic Danijela, and Hackenberg Michael. 2014. “sRNAbench: profiling of small RNAs and its sequence variants in single or multi-species high-throughput experiments.” <em>Methods in Next Generation Sequencing</em> 1 (1): 2084–7173. doi:<a href="https://doi.org/10.2478/mngs-2014-0001">10.2478/mngs-2014-0001</a>.</p>
</div>
<div id="ref-emarti">
<p>Martí, Eulàlia, Lorena Pantano, Mónica Bañez-Coronel, Franc Llorens, Elena Miñones-Moyano, Sílvia Porta, Lauro Sumoy, Isidre Ferrer, and Xavier Estivill. 2010. “A myriad of miRNA variants in control and Huntington’s disease brain regions detected by massively parallel sequencing.” <em>Nucleic Acids Res.</em> 38: 7219–35. doi:<a href="https://doi.org/10.1093/nar/gkq575">10.1093/nar/gkq575</a>.</p>
</div>
<div id="ref-morin">
<p>Morin, R. D., M. D. O’Connor, M. Griffith, F. Kuchenbauer, A. Delaney, A.-L. Prabhu, Y. Zhao, et al. 2008. “Application of massively parallel sequencing to microRNA profiling and discovery in human embryonic stem cells.” <em>Genome Res.</em> 18: 610–21. doi:<a href="https://doi.org/10.1101/gr.7179508">10.1101/gr.7179508</a>.</p>
</div>
<div id="ref-isomirs">
<p>Pantano L, Estivil X, Marti E. 2010. “SeqBuster.” <em>Nucleic Acids Res.</em> 38: e34. doi:<a href="https://doi.org/10.1093/nar/gkp1127">10.1093/nar/gkp1127</a>.</p>
</div>
<div id="ref-perezenciso">
<p>Pérez-Enciso, Miguel, and Michel Tenenhaus. 2003. “Prediction of clinical outcome with microarray data: a partial least squares discriminant analysis (PLS-DA) approach.” <em>Human Genetics</em> 112: 581–92. doi:<a href="https://doi.org/10.1007/s00439-003-0921-9">10.1007/s00439-003-0921-9</a>.</p>
</div>
<div id="ref-xia">
<p>Xia, Jianguo, and David S Wishart. 2011. “Web-based inference of biological patterns, functions and pathways from metabolomic data using MetaboAnalyst.” <em>Nature Protocols</em> 6: 743–60. doi:<a href="https://doi.org/10.1038/nprot.2011.319">10.1038/nprot.2011.319</a>.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#citing-isomirs">Citing isomiRs</a></li>
      <li><a href="#input-format">Input format</a></li>
      <li><a href="#isomirdataseq-class">IsomirDataSeq class</a></li>
      <li><a href="#reading-input">Reading input</a></li>
      <li><a href="#descriptive-analysis">Descriptive analysis</a></li>
      <li><a href="#count-data">Count data</a></li>
      <li><a href="#supervised-classification">Supervised classification</a></li>
      <li><a href="#gene---mirna-integration">Gene - miRNA integration</a></li>
      <li><a href="#session-info">Session info</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Lorena Pantano, Georgia Escaramis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
