<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Create count matrix with different summarizing options — isoCounts • isomiRs</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">isomiRs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/isomiRs.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create count matrix with different summarizing options</h1>
    </div>

    
    <p>This function collapses isomiRs into different groups. It is a similar
concept than how to work with gene isoforms. With this function,
different changes can be put together into a single miRNA variant.
For instance all sequences with variants at 3' end can be
considered as different elements in the table
or analysis having the following naming
<code>hsa-miR-124a-5p.iso.t3:AAA</code>.</p>
    

    <pre class="usage"><span class='fu'>isoCounts</span>(<span class='no'>ids</span>, <span class='kw'>ref</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>iso5</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>iso3</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>add</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>subs</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>seed</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>minc</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>mins</span> <span class='kw'>=</span> <span class='fl'>1</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ids</th>
      <td><p>Object of class <a href='IsomirDataSeq.html'>IsomirDataSeq</a>.</p></td>
    </tr>
    <tr>
      <th>ref</th>
      <td><p>Differentiate reference miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>iso5</th>
      <td><p>Differentiate trimming at 5 miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>iso3</th>
      <td><p>Differentiate trimming at 3 miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>add</th>
      <td><p>Differentiate additions miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>subs</th>
      <td><p>Differentiate nt substitution miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>seed</th>
      <td><p>Differentiate changes in 2-7 nts from rest.</p></td>
    </tr>
    <tr>
      <th>minc</th>
      <td><p>Int minimum number of isomiR sequences to be included.</p></td>
    </tr>
    <tr>
      <th>mins</th>
      <td><p>Int minimum number of samples with number of
sequences bigger than <code>minc</code> counts.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><a href='IsomirDataSeq.html'>IsomirDataSeq</a> object with new count table.
The count matrix can be access with <code><a href='counts.html'>counts(ids)</a></code>.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>You can merge all isomiRs into miRNAs by calling the function only
with the first parameter <code>isoCounts(ids)</code>.
You can get a table with isomiRs altogether and
the reference miRBase sequences by calling the function with <code>ref=TRUE</code>.
You can get a table with 5' trimming isomiRS, miRBase reference and
the rest by calling with <code>isoCounts(ids, ref=TRUE, iso5=TRUE)</code>.
If you set up all parameters to TRUE, you will get a table for
each different sequence mapping to a miRNA (i.e. all isomiRs).</p>
<p>Examples for the naming used for the isomiRs are at
http://seqcluster.readthedocs.org/mirna_annotation.html#mirna-annotation.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>data</span>(<span class='no'>mirData</span>)
<span class='no'>ids</span> <span class='kw'>&lt;-</span> <span class='fu'>isoCounts</span>(<span class='no'>mirData</span>, <span class='kw'>ref</span><span class='kw'>=</span><span class='fl'>TRUE</span>)
<span class='fu'>head</span>(<span class='fu'><a href='counts.html'>counts</a></span>(<span class='no'>ids</span>))</div><div class='output co'>#&gt;                        cc1    cc2    cc3    ct2    ct3    ct4
#&gt; hsa-let-7a-2-3p.iso      2      7      2      0      2      6
#&gt; hsa-let-7a-2-3p.ref      3      6      2      4      0      6
#&gt; hsa-let-7a-3p.iso      653    593    543    448    312    553
#&gt; hsa-let-7a-3p.ref      114    114     87     48     31     82
#&gt; hsa-let-7a-5p.iso    81905  73478  54347  58497  40218  52069
#&gt; hsa-let-7a-5p.ref   235825 171354 149541 163569 114028 123454</div><div class='input'><span class='co'># taking into account isomiRs and reference sequence.</span>
<span class='no'>ids</span> <span class='kw'>&lt;-</span> <span class='fu'>isoCounts</span>(<span class='no'>mirData</span>, <span class='kw'>ref</span><span class='kw'>=</span><span class='fl'>TRUE</span>, <span class='kw'>minc</span><span class='kw'>=</span><span class='fl'>10</span>, <span class='kw'>mins</span><span class='kw'>=</span><span class='fl'>6</span>)
<span class='fu'>head</span>(<span class='fu'><a href='counts.html'>counts</a></span>(<span class='no'>ids</span>))</div><div class='output co'>#&gt;                      cc1    cc2    cc3    ct2    ct3    ct4
#&gt; hsa-let-7a-3p.iso    653    593    543    448    312    553
#&gt; hsa-let-7a-3p.ref    114    114     87     48     31     82
#&gt; hsa-let-7a-5p.iso  81905  73478  54347  58497  40218  52069
#&gt; hsa-let-7a-5p.ref 235825 171354 149541 163569 114028 123454
#&gt; hsa-let-7b-3p.iso   1033   1729    668    808    483   1058
#&gt; hsa-let-7b-5p.iso  31209  36938  18988  22156  14187  20585</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Lorena Pantano, Georgia Escaramis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
