<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Create count matrix with different summarizing options — isoCounts • isomiRs</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Create count matrix with different summarizing options — isoCounts" />

<meta property="og:description" content="This function collapses isomiRs into different groups. It is a similar
concept than how to work with gene isoforms. With this function,
different changes can be put together into a single miRNA variant.
For instance all sequences with variants at 3' end can be
considered as different elements in the table
or analysis having the following naming
hsa-miR-124a-5p.iso.t3:AAA." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">isomiRs</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.11.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/isomiRs.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Create count matrix with different summarizing options</h1>
    
    <div class="hidden name"><code>isoCounts.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function collapses isomiRs into different groups. It is a similar
concept than how to work with gene isoforms. With this function,
different changes can be put together into a single miRNA variant.
For instance all sequences with variants at 3' end can be
considered as different elements in the table
or analysis having the following naming
<code>hsa-miR-124a-5p.iso.t3:AAA</code>.</p>
    
    </div>

    <pre class="usage"><span class='fu'>isoCounts</span>(<span class='no'>ids</span>, <span class='kw'>ref</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>iso5</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>iso3</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>add</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>snv</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>seed</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>all</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>minc</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>mins</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>merge_by</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ids</th>
      <td><p>Object of class <a href='IsomirDataSeq.html'>IsomirDataSeq</a>.</p></td>
    </tr>
    <tr>
      <th>ref</th>
      <td><p>Differentiate reference miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>iso5</th>
      <td><p>Differentiate trimming at 5 miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>iso3</th>
      <td><p>Differentiate trimming at 3 miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>add</th>
      <td><p>Differentiate additions miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>snv</th>
      <td><p>Differentiate nt substitution miRNA from rest.</p></td>
    </tr>
    <tr>
      <th>seed</th>
      <td><p>Differentiate changes in 2-7 nts from rest.</p></td>
    </tr>
    <tr>
      <th>all</th>
      <td><p>Differentiate all isomiRs.</p></td>
    </tr>
    <tr>
      <th>minc</th>
      <td><p>Int minimum number of isomiR sequences to be included.</p></td>
    </tr>
    <tr>
      <th>mins</th>
      <td><p>Int minimum number of samples with number of
sequences bigger than <code>minc</code> counts.</p></td>
    </tr>
    <tr>
      <th>merge_by</th>
      <td><p>Column in coldata to merge samples into a single
column in counts. Useful to combine technical replicates.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><a href='IsomirDataSeq.html'>IsomirDataSeq</a> object with new count table.
The count matrix can be access with <code><a href='counts.html'>counts(ids)</a></code>.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>You can merge all isomiRs into miRNAs by calling the function only
with the first parameter <code>isoCounts(ids)</code>.
You can get a table with isomiRs altogether and
the reference miRBase sequences by calling the function with <code>ref=TRUE</code>.
You can get a table with 5' trimming isomiRS, miRBase reference and
the rest by calling with <code>isoCounts(ids, ref=TRUE, iso5=TRUE)</code>.
If you set up all parameters to TRUE, you will get a table for
each different sequence mapping to a miRNA (i.e. all isomiRs).</p>
<p>Examples for the naming used for the isomiRs are at
http://seqcluster.readthedocs.org/mirna_annotation.html#mirna-annotation.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>mirData</span>)
<span class='no'>ids</span> <span class='kw'>&lt;-</span> <span class='fu'>isoCounts</span>(<span class='no'>mirData</span>, <span class='kw'>ref</span><span class='kw'>=</span><span class='fl'>TRUE</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='fu'><a href='counts.html'>counts</a></span>(<span class='no'>ids</span>))</div><div class='output co'>#&gt;                        cc1    cc2    cc3    cc4    cc5    cc6    cc7    ct1
#&gt; hsa-let-7a-2-3p          2      7      2      2      6      0      3      0
#&gt; hsa-let-7a-2-3p;ref      3      6      2      7      5      6      4      0
#&gt; hsa-let-7a-3p          653    593    543    530    594    335    574    219
#&gt; hsa-let-7a-3p;ref      114    114     87     79    137     54    107     39
#&gt; hsa-let-7a-5p        81905  73478  54347  80277  75900  45619  67502  65481
#&gt; hsa-let-7a-5p;ref   235825 171354 149541 180654 168884 107430 153061 143030
#&gt;                        ct2    ct3    ct4    ct5    ct6    ct7
#&gt; hsa-let-7a-2-3p          0      2      6      6      0      0
#&gt; hsa-let-7a-2-3p;ref      4      0      6      6      8      3
#&gt; hsa-let-7a-3p          448    312    553    510    403    443
#&gt; hsa-let-7a-3p;ref       48     31     82    112     49     76
#&gt; hsa-let-7a-5p        58497  40218  52069  56641  71781  59651
#&gt; hsa-let-7a-5p;ref   163569 114028 123454 133092 158909 140272</div><div class='input'><span class='co'># taking into account isomiRs and reference sequence.</span>
<span class='no'>ids</span> <span class='kw'>&lt;-</span> <span class='fu'>isoCounts</span>(<span class='no'>mirData</span>, <span class='kw'>ref</span><span class='kw'>=</span><span class='fl'>TRUE</span>, <span class='kw'>minc</span><span class='kw'>=</span><span class='fl'>10</span>, <span class='kw'>mins</span><span class='kw'>=</span><span class='fl'>6</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='fu'><a href='counts.html'>counts</a></span>(<span class='no'>ids</span>))</div><div class='output co'>#&gt;                      cc1    cc2    cc3    cc4    cc5    cc6    cc7    ct1
#&gt; hsa-let-7a-3p        653    593    543    530    594    335    574    219
#&gt; hsa-let-7a-3p;ref    114    114     87     79    137     54    107     39
#&gt; hsa-let-7a-5p      81905  73478  54347  80277  75900  45619  67502  65481
#&gt; hsa-let-7a-5p;ref 235825 171354 149541 180654 168884 107430 153061 143030
#&gt; hsa-let-7b-3p       1033   1729    668   1351   1741    644   1326    335
#&gt; hsa-let-7b-3p;ref      4    220     11     34    143     53    173      3
#&gt;                      ct2    ct3    ct4    ct5    ct6    ct7
#&gt; hsa-let-7a-3p        448    312    553    510    403    443
#&gt; hsa-let-7a-3p;ref     48     31     82    112     49     76
#&gt; hsa-let-7a-5p      58497  40218  52069  56641  71781  59651
#&gt; hsa-let-7a-5p;ref 163569 114028 123454 133092 158909 140272
#&gt; hsa-let-7b-3p        808    483   1058   1051   1253    890
#&gt; hsa-let-7b-3p;ref    123     11     91    143    178     98</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Lorena Pantano, Georgia Escaramis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

