<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Loads miRNA annotation from seqbuster tool or pre-processed data. — IsomirDataSeqFromFiles • isomiRs</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Loads miRNA annotation from seqbuster tool or pre-processed data. — IsomirDataSeqFromFiles" />

<meta property="og:description" content="This function parses
output of seqbuster tool to allow isomiRs/miRNAs analysis of samples
in different groups such as
characterization, differential expression and clustering. It creates an
isomiRs::IsomirDataSeq object." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">isomiRs</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.11.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/isomiRs.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Loads miRNA annotation from seqbuster tool or pre-processed data.</h1>
    
    <div class="hidden name"><code>IsomirDataSeqFromFiles.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function parses
output of seqbuster tool to allow isomiRs/miRNAs analysis of samples
in different groups such as
characterization, differential expression and clustering. It creates an
<a href='IsomirDataSeq.html'>isomiRs::IsomirDataSeq</a> object.</p>
    
    </div>

    <pre class="usage"><span class='fu'>IsomirDataSeqFromFiles</span>(<span class='no'>files</span>, <span class='no'>coldata</span>, <span class='kw'>rate</span> <span class='kw'>=</span> <span class='fl'>0.2</span>, <span class='kw'>canonicalAdd</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>uniqueMism</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>uniqueHits</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>design</span> <span class='kw'>=</span> ~<span class='fl'>1L</span>,
  <span class='kw'>minHits</span> <span class='kw'>=</span> <span class='fl'>1L</span>, <span class='kw'>header</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>skip</span> <span class='kw'>=</span> <span class='fl'>0</span>, <span class='kw'>quiet</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='no'>...</span>)

<span class='fu'>IsomirDataSeqFromRawData</span>(<span class='no'>rawdata</span>, <span class='no'>coldata</span>, <span class='kw'>design</span> <span class='kw'>=</span> ~<span class='fl'>1L</span>, <span class='kw'>pct</span> <span class='kw'>=</span> <span class='fl'>0.1</span>,
  <span class='kw'>n_snv</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>whitelist</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>files</th>
      <td><p>files with the output of seqbuster tool</p></td>
    </tr>
    <tr>
      <th>coldata</th>
      <td><p>data frame containing groups for each sample</p></td>
    </tr>
    <tr>
      <th>rate</th>
      <td><p>minimum counts fraction to consider a mismatch a real mutation</p></td>
    </tr>
    <tr>
      <th>canonicalAdd</th>
      <td><p><code>boolean</code> only keep A/T non-template addition.
All non-template nucleotides at the 3' end will be removed if they
contain C/G nts.</p></td>
    </tr>
    <tr>
      <th>uniqueMism</th>
      <td><p><code>boolean</code> only keep mutations that have
a unique hit to one miRNA molecule. For instance, if the sequence map
to two different miRNAs, then it would be removed.</p></td>
    </tr>
    <tr>
      <th>uniqueHits</th>
      <td><p><code>boolean</code> whether filtering ambigous sequences or not.</p></td>
    </tr>
    <tr>
      <th>design</th>
      <td><p>a <code>formula</code> to pass to <a href='https://www.rdocumentation.org/packages/DESeq2/topics/DESeqDataSet'>DESeq2::DESeqDataSet</a></p></td>
    </tr>
    <tr>
      <th>minHits</th>
      <td><p>Minimum number of reads in the sample to consider it
in the final matrix.</p></td>
    </tr>
    <tr>
      <th>header</th>
      <td><p>boolean to indicate files contain headers</p></td>
    </tr>
    <tr>
      <th>skip</th>
      <td><p>skip first line when reading files</p></td>
    </tr>
    <tr>
      <th>quiet</th>
      <td><p>boolean indicating to print messages
while reading files. Default <code>FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>arguments provided to
<code>SummarizedExperiment</code>.
including rowData.</p></td>
    </tr>
    <tr>
      <th>rawdata</th>
      <td><p>data.frame stored in metadata slot of <a href='IsomirDataSeq.html'>IsomirDataSeq</a> object.</p></td>
    </tr>
    <tr>
      <th>pct</th>
      <td><p>numeric used to remove isomiRs with an importance lower than
this value. Importance is calculated by dividing the isomiR count
by the total counts of the miRNA to which it maps.</p></td>
    </tr>
    <tr>
      <th>n_snv</th>
      <td><p>numeric used to remove isomiRs with more than this number of
single nucleotide variants (indels are counted here).</p></td>
    </tr>
    <tr>
      <th>whitelist</th>
      <td><p>character vector with sequences to keep even
if the filtering step would have removed them. They have to match
the <code>seq</code> column in the table.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><a href='IsomirDataSeq.html'>IsomirDataSeq</a> class object.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function parses the output of
http://seqcluster.readthedocs.org/mirna_annotation.html
for each sample to create a count matrix for isomiRs, miRNAs or
isomiRs grouped in
types (i.e all sequences with variations at 5' but ignoring any other type).
It creates
<a href='IsomirDataSeq.html'>isomiRs::IsomirDataSeq</a> object (see link to example usage of
this class)
to allow visualization, queries, differential
expression analysis and clustering.
To create the <a href='IsomirDataSeq.html'>isomiRs::IsomirDataSeq</a>, it parses the isomiRs
files, and generates
an initial matrix having all isomiRs detected among samples. As well,
it creates
a summary for each isomiR type (trimming, addition and substitution) to
visualize general isomiRs distribution.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>path</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/system.file'>system.file</a></span>(<span class='st'>"extra"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"isomiRs"</span>)
<span class='no'>fn_list</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list.files'>list.files</a></span>(<span class='no'>path</span>, <span class='kw'>full.names</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='no'>de</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/data.frame'>data.frame</a></span>(<span class='kw'>row.names</span><span class='kw'>=</span><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"f1"</span> , <span class='st'>"f2"</span>),
                 <span class='kw'>condition</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"newborn"</span>, <span class='st'>"newborn"</span>))
<span class='no'>ids</span> <span class='kw'>&lt;-</span> <span class='fu'>IsomirDataSeqFromFiles</span>(<span class='no'>fn_list</span>, <span class='kw'>coldata</span><span class='kw'>=</span><span class='no'>de</span>)</div><div class='output co'>#&gt; <span class='error'>Error in guess_header_(datasource, tokenizer, locale): Cannot read file /Users/lpantano/repos/isomiRs/inst/extra/mirtop: Invalid argument</span></div><div class='input'>
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='fu'><a href='counts.html'>counts</a></span>(<span class='no'>ids</span>))</div><div class='output co'>#&gt; <span class='error'>Error in counts(ids): object 'ids' not found</span></div><div class='input'><span class='fu'>IsomirDataSeqFromRawData</span>(<span class='fu'>metadata</span>(<span class='no'>ids</span>)<span class='kw'>[[</span><span class='st'>"rawData"</span>]], <span class='no'>de</span>)</div><div class='output co'>#&gt; <span class='error'>Error in metadata(ids): object 'ids' not found</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Lorena Pantano, Georgia Escaramis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

