<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Clustering miRNAs-genes pairs in similar pattern expression — isoNetwork • isomiRs</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">isomiRs</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Clustering miRNAs-genes pairs in similar pattern expression</h1>
    </div>

    
    <p>Clustering miRNAs-genes pairs</p>
    

    <pre class="usage"><span class='fu'>isoNetwork</span>(<span class='no'>mirna_rse</span>, <span class='no'>gene_rse</span>, <span class='no'>target</span>, <span class='no'>org</span>, <span class='kw'>summarize</span> <span class='kw'>=</span> <span class='st'>"group"</span>,
  <span class='kw'>genename</span> <span class='kw'>=</span> <span class='st'>"ENSEMBL"</span>, <span class='kw'>min_cor</span> <span class='kw'>=</span> -<span class='fl'>0.6</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>mirna_rse</th>
      <td><p><a href='http://www.rdocumentation.org/packages/SummarizedExperiment/topics/RangedSummarizedExperiment-class'>SummarizedExperiment::SummarizedExperiment</a> with miRNA
information. See details.</p></td>
    </tr>
    <tr>
      <th>gene_rse</th>
      <td><p><a href='http://www.rdocumentation.org/packages/SummarizedExperiment/topics/RangedSummarizedExperiment-class'>SummarizedExperiment::SummarizedExperiment</a> with gene
information. See details.</p></td>
    </tr>
    <tr>
      <th>target</th>
      <td><p>Matrix with miRNAs (columns) and genes (rows)
target prediction (1 if it is a target, 0 if not).</p></td>
    </tr>
    <tr>
      <th>org</th>
      <td><p><a href='http://www.rdocumentation.org/packages/AnnotationDbi/topics/AnnotationDb-class'>AnnotationDbi::AnnotationDb</a> obejct. For example:(org.Mm.eg.db).</p></td>
    </tr>
    <tr>
      <th>summarize</th>
      <td><p>Character column name in <code>colData(rse)</code> to use to group
samples and compare betweem miRNA/gene expression.</p></td>
    </tr>
    <tr>
      <th>genename</th>
      <td><p>Character keytype of the gene
names in gene_rse object.</p></td>
    </tr>
    <tr>
      <th>min_cor</th>
      <td><p>Numeric cutoff to consider a miRNA to regulate a target.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>list with network information</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This function will correlate miRNA and gene expression data using
a specific metadata variable to group samples and detect pattern
of expression that will be annotated with GO terms.
mirna_rse and gene_rse can be created using the following code:
    <code>mi_rse = SummarizedExperiment(assays=SimpleList(norm=mirna_matrix), colData, metadata=list(sign=mirna_keep))</code></p>
<p>where, <code>mirna_matrix</code> is the normalized counts expression,
<code>colData</code> is the metadata information and <code>mirna_keep</code>
the list of miRNAs to be used by this function.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>library</span>(<span class='no'>org.Mm.eg.db</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: AnnotationDbi</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘AnnotationDbi’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:isomiRs’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    select</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:dplyr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    select</span></div><div class='output co'>#&gt; </div><div class='input'><span class='fu'>library</span>(<span class='no'>clusterProfiler</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: DOSE</span></div><div class='output co'>#&gt; </div><div class='output co'>#&gt; <span class='message'>DOSE v3.3.2  For help: https://guangchuangyu.github.io/DOSE</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>If you use DOSE in published research, please cite:</span>
#&gt; <span class='message'>Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</span></div><div class='output co'>#&gt; <span class='message'>clusterProfiler v3.5.6  For help: https://guangchuangyu.github.io/clusterProfiler</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>If you use clusterProfiler in published research, please cite:</span>
#&gt; <span class='message'>Guangchuang Yu., Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘clusterProfiler’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:purrr’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    simplify</span></div><div class='input'><span class='fu'>data</span>(<span class='no'>isoExample</span>)
<span class='co'># ego &lt;- enrichGO(row.names(assay(gene_ex_rse, "norm")),</span>
<span class='co'>#                 org.Mm.eg.db, "ENSEMBL", ont = "BP")</span>
<span class='co'># data = isoNetwork(mirna_ex_rse, gene_ex_rse, ma_ex,</span>
<span class='co'>#                   org = slot(ego, "result"))</span>
<span class='co'># isoPlotNet(data)</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Lorena Pantano, Georgia Escaramis, Christos Argyropoulos.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
